src/App.js

```diff
@@ -1,26 +1,47 @@
 import React from 'react';
-import logo from './logo.svg';
+import { Container } from 'react-bootstrap'
+import { ConnectedRouter } from 'connected-react-router'
+import {Switch, Route} from 'react-router-dom';
+
+import {useSelector} from "react-redux";
+import { history } from './store'
+import {userInfoSelector} from "./state/user/user.selectors";
+
+import NavBar from './components/NavBar';
+import Login from './components/Login';
+import SignUp from "./components/Signup";
+import ForgotPassword from "./components/ForgotPassword";
+import Profile from "./components/Profile";
+import ChangePassword from "./components/ChangePassword";
+import Landing from "./components/Landing";
+
 import './App.css';
 
+
+/**
+ * Main Component of the Application - Routes to specific components based on the Path.
+ */
 function App() {
+  const user = useSelector(userInfoSelector);
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <NavBar user={user} />
+      <Container className={'main-container'}>
+        <Container style={{justifyContent: 'center', margin: 'unset'}} className={'app-container'}>
+          <ConnectedRouter history={history}>
+            <Switch>
+              <Route path="/login" exact component={ Login} />
+              <Route path="/signup" exact component={ SignUp} />
+              <Route path="/forgotpassword" exact component={ ForgotPassword} />
+              <Route path="/changepassword" exact component={ ChangePassword} />
+              <Route path="/profile" exact component={ Profile} />
+              <Route path="/dashboard" exact component={ Landing} />
+            </Switch>
+          </ConnectedRouter>
+        </Container>
+      </Container>`
+    </>
```

Comment:
@nikhilrayaprolu It'd be good to have default route "\" that redirects to the correct component.
If isAuthenticated, then redirects to login,
Otherwise, redirects to dashboard.

And these two issues should be fixed as well.
 - After login, users can input '\login'  (should redirect to '\dashboard')
 - Before login, users can input '\dashboard' (should redirect to '\login')